crates/matrix-sdk-crypto/src/session_manager/group_sessions.rs

```diff
@@ -436,13 +436,27 @@ impl GroupSessionManager {
                         shared.difference(&recipient_device_ids).collect::<BTreeSet<_>>();
 
                     should_rotate = !newly_deleted_or_blacklisted.is_empty();
+                    if should_rotate {
+                        debug!(
+                            "collect_session_recipients: {} rotating due to these devices being deleted/blacklisted {:?}",
+                            outbound.session_id(),
```

Comment:
The session ID is already attached to the parent span.

Reply:
Right you are, removing.